usefulUtilities/hiFiCalendar/entityServerScripts/hiFiCalendarServer.js

```diff
@@ -8,119 +8,348 @@
 
 
 (function() {
+    var INTERVAL_FREQUENCY_MS = 60000;
+    var EXPIRY_BUFFER = 300000;
+    var HOURS_PER_DAY = 24;
+    var NOON = 12;
+    var CHANNEL = "HiFi.Google.Calendar";
+    var GREEN_TEXT = [168, 255, 168];
+    var RED_TEXT = [255, 168, 168];
+    var GREEN = [125, 255, 125];
+    var RED = [255, 125, 125];
+
     var that;
 
     this.remotelyCallable = [
+        "refreshToken",
         "enteredMeetingZone",
-        "leftMeetingZone",
-        "updateSignColor",
-        "showNoScheduledEvents",
-        "addEvent",
-        "clearEventList"
+        "leftMeetingZone"
     ];
 
 
     this.preload = function(entityID) {
         that = this;
         that.entityID = entityID;
-        that.entityProperties = Entities.getEntityProperties(that.entityID, ['id', 'name', 'type', 'parentID']);
-        that.room = {};
-        if (that.entityProperties.name.indexOf("_SCHEDULE") === -1 && 
-            that.entityProperties.name.indexOf("_OCCUPANTS") === -1 && 
-            that.entityProperties.type === "Text") {
+        that.entityProperties = Entities.getEntityProperties(that.entityID, ['userData']);
+        if (that.entityProperties.userData.length !== 0) {
+            try {
+                that.userData = JSON.parse(that.entityProperties.userData);
```

Comment:
why can't you do `var userData` instead of `that.userData` ? i don't think you want to be making `userData` an object inside `that`, do you?

Reply:
I was concerned that creating var userData would affect other entities running this script.